plasmapy/analysis/swept_langmuir/floating_potential.py

```diff
@@ -78,71 +115,73 @@ def find_floating_potential(
-        crossing-islands.
-
-    indices: `slice`
-        A `slice` object representing the indices of ``voltage`` and ``current``
-        arrays used for the fit.
+        The calculated floating potential (same units as the
+        ``voltage`` array).  Returns `numpy.nan` if the floating
+        potential can not be determined.
+
+    extras: `VFExtras`
+        Additional information from the fit:
+
+        ``extras.vf_err`` (`float` or `numpy.nan`)
+            The uncertainty associated with the floating potential
+            calculation (units same as ``vf``).  Returns `numpy.nan`
+            if the floating potential can not be determined.  Like
+            :math:`V_f`:, the calculation depends on the applied fit
+            function.  The ``root_solve()`` method also describes how
+            this is calculated.
+
+        ``extras.rsq`` (`float`)
+            The coefficient of determination (r-squared) value of the
+            fit.  See the documentation of the ``rsq`` property on the
+            associated fit function (e.g. the
+            `~plasmapy.analysis.fit_functions.ExponentialPlusOffset.rsq`
+            property of
+            `~plasmapy.analysis.fit_functions.ExponentialPlusOffset`).
+
+        ``extras.fitted_func`` (:term:`fit-function`)
+            The computed :term:`fit-function` specified by ``fit_type``.
+
+        ``extras.islands`` (``List[slice]``)
+            List of `slice` objects representing the indices of the
+            identified crossing-islands.
+
+        ``extras.fitted_indices`` (`slice`)
+            A `slice` object representing the indices of the ``voltage``
+            and ``current`` arrays used for the fit.
 
     Notes
     -----
     The internal functionality works like:
 
-    1. The current array ``current`` is scanned for all points equal to zero and
-       point pairs that straddle :math:`I = 0`.  This forms an array of
-       "crossing-points."
-    2. The crossing-points are then grouped into "crossing-islands" in based on
-       the ``threshold`` keyword.
-
-       - A new island is formed when a successive crossing-point is more (index)
-         steps away from the previous crossing-point than allowed by
-         ``threshold``.
-       - If multiple crossing-islands are identified, then the span from the
-         first point in the first island to the last point in the last island is
-         compared to ``min_points``.  If the span is less than or equal to
-         ``min_points``, then that span is taken as one larger crossing-island
-         for the fit; otherwise, the function is incapable of identifying
-         :math:`V_f` and will return `numpy.nan` values.
+    1. The current array ``current`` is scanned for all points equal to
+       zero and point pairs that straddle :math:`I = 0`.  This forms an
+       array of "crossing-points."
+    2. The crossing-points are then grouped into "crossing-islands" in
```

Comment:
[out-of-scope and optional] I'm wondering if it would be helpful to add "crossing-point" and "crossing-island" to the glossary. Definitely not necessary, though.

Reply:
... interesting thought. Not sure.  If they become more ubiquitous in the package and in the documentation, then yes.  For now, I'll let it be.